### PR TITLE
GH Actions: use ACCESS_TOKEN_GRADLE_BUILD for building VRO

### PR DESCRIPTION
**Pull request by yoomlam**
## What was the problem?

[Remove references to yoomlam's GH PAT](https://github.com/department-of-veterans-affairs/abd-vro/pull/925#top) #925 did **not** work as desired. It appeared to work because GH Action workflows were using [a Gradle cache](https://github.com/department-of-veterans-affairs/abd-vro/blob/011a1b4452f824b9ee43f2ef7577cfe04208287e/.github/actions/setup-vro/action.yml#L19), hence it didn't need to [use the access token](https://github.com/department-of-veterans-affairs/abd-vro/blob/011a1b4452f824b9ee43f2ef7577cfe04208287e/settings.gradle#L21) to download [the assets](https://github.com/department-of-veterans-affairs/abd-vro/blob/011a1b4452f824b9ee43f2ef7577cfe04208287e/build.gradle#L3-L4).

A Personal Access Token is needed by Gradle to download the assets. Using `secrets.GITHUB_TOKEN` does not work, as demonstrated in [temporarily disable gradle caching](https://github.com/department-of-veterans-affairs/abd-vro/pull/989/commits/3cee2c126ce002a12bab4151c572b1204acea58e) and the [Test code GH workflow](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4000155424/jobs/6864934964#step:5:57).
[LHDI docs say](https://animated-carnival-57b3e7f5.pages.github.io/starterkits/java/getting-started/) to create a PAT.

## How does this fix it?[^1]

Use `ACCESS_TOKEN_GRADLE_BUILD` as the access token for building VRO. Basically revert #925, but use the more specific `ACCESS_TOKEN_GRADLE_BUILD` variable name. The access token was generated using the machine user `abd-vro-machine`'s account.

But continue to use `secrets.GITHUB_TOKEN` for publishing images to GHCR to comply with SecRel requirements.

## How to test this PR

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.

### CODE REVIEW

**Comment by yoomlam** [comment]
The PR checks above ran without gradle caching, so remember to re-enable it before merging this PR.

**Comment by yoomlam** [comment]
Manually ran these workflows:
* [SBOM](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4000342707)
* [Trivy](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4000348230)
* [Snyk](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4000354298)
* [Deploy to Dev](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4000415259)

which all succeeded.

File: .github/workflows/secrel4prs.yml [hunk 1]
```diff
@@ -59,7 +59,7 @@ jobs:
         uses: ./.github/actions/publish-images
         with:
           target_env: ${{ env.TARGET_ENV }}
-          access_token: ${{ secrets.GITHUB_TOKEN }}
+          access_token: ${{ secrets.ACCESS_TOKEN_GRADLE_BUILD }}
```
Comment (yoomlam) [inline .github/workflows/secrel4prs.yml:62]: Tested [here](https://github.com/department-of-veterans-affairs/abd-vro-internal/actions/runs/4000430160) -- I'll continue to tweak this in PR #963, so it's okay if this fails.

File: .github/actions/setup-vro/action.yml [hunk 2]
```diff
@@ -15,16 +15,16 @@ runs:
...
+#    - name: "Setup Gradle"
+#      uses: gradle/gradle-build-action@v2
+#      with:
+#        # Only write to the cache for builds on the 'main' and 'develop' branches. (Default is 'main' only.)
+#        # Builds on other branches will only read existing entries from the cache.
+#        cache-read-only: ${{ github.ref != 'refs/heads/main' && github.ref != 'refs/heads/qa' && github.ref != 'refs/heads/develop' }}
+#        # To avoid a growing cache over time, attempt to delete any files in the Gradle User Home
+#        # that were not used by Gradle during the workflow, prior to saving the cache.
+#        # https://github.com/gradle/gradle-build-action#removing-unused-files-from-gradle-user-home-before-saving-to-cache
+#        gradle-home-cache-cleanup: true
```
Comment (yoomlam) [inline .github/actions/setup-vro/action.yml:27]: I'll revert these changes to re-enable gradle caching before merging this PR.

File: .github/actions/publish-images/action.yml [hunk 3]
```diff
@@ -35,7 +35,7 @@ runs:
     - name: "Test code"
       if: inputs.run_tests == 'true'
       shell: bash
-      run: ./gradlew check
+      run: ./gradlew check -PGITHUB_ACCESS_TOKEN=${{ inputs.access_token }}
```
Comment (yoomlam) [inline .github/actions/publish-images/action.yml:38]: To address [this error](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4000415259/jobs/6865654740#step:5:621).

**Comment by yoomlam** [comment]
* [Deploy to Dev](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4000657607)

File: .github/workflows/secrel.yml [hunk 4]
```diff
@@ -13,7 +13,10 @@ on:
   workflow_dispatch:
 
 env:
-  GITHUB_ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
+  # If SecRel can't push signed images, revert to use `secrets.ACCESS_TOKEN` instead.
+  # secrets.ACCESS_TOKEN_GRADLE_BUILD only has `read:packages` permission
+  # while secrets.ACCESS_TOKEN has `repo` and `write:packages` permission
+  GITHUB_ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN_GRADLE_BUILD }}
```
Comment (yoomlam) [inline .github/workflows/secrel.yml:19]: @sethdarragile6 FYI

Comment (yoomlam) [reply .github/workflows/secrel.yml:19]: Once SecRel stop failing, we should test this without Gradle caching to ensure this `ACCESS_TOKEN_GRADLE_BUILD` is sufficient and then remove the comment and `secrets.ACCESS_TOKEN` from https://github.com/department-of-veterans-affairs/abd-vro-internal/settings/secrets/actions

**Review by sethdarragile6 — APPROVED** [review]
lgtm

**Pull request merged**